lst = ["ножницы", "карандаш", "яблоко", "книга"]

# получаем итератор для итерируемого объекта
iterator = iter(lst)

try:
    while True:
        next_val = next(iterator)
        print('Очередное значение:', next_val)
except StopIteration:
    # явно напечатаем сообщение об окончании итерации,
    # хотя цикл for этого не делает и ошибка просто подавляется
    print('Итерация закончена')
print('Программа завершена')